cpp/include/cuspatial/experimental/allpairs_multipoint_equals_count.cuh

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <iterator>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute the number of multipoint pairs that are equal.
```

Comment:
Something still sticks out from this docstring so I'll say it out loud:

1. There's nothing in the docstring to show that `lhs` is the range of elements to search and `rhs` is the range to search from. And nothing to show this operation is non-commutative (directed).

Reply:
This one got less love than the .hpp file. I've fleshed it out some more and put in a few good examples.